### DISCUSSION TITLE
2FA for Specific User Group(s)

### DISCUSSION THREAD

**Discussion by jsurp2001** [codeigniter4/shield#1120] [Ideas]
I feel 2FA for a specific group or groups would be a nice feature to put in place, especially for Admin and Superadmin groups but not necessarily everyone. Thoughts?

**Comment by datamweb** [comment]
Currently, this is not possible by default. I already posted something similar on #324 but got no feedback.
Now if you need such a feature you have to write your own custom action. Refer to [defining-new-actions](https://shield.codeigniter.com/references/authentication/auth_actions/?h=action#defining-new-actions).

Later requests were raised about this, personally I think that these features should be present in the Shield by default. Activating actions for a specific user, or a specific group and...

**Reply by jsurp2001** [reply]
Thank you!

**Comment by datamweb** [comment]
The requested functionality has been added in PR #1328 .
